.gitignore

```diff
@@ -14,6 +14,7 @@
 build
 coverage.xml
 dist
+primevue-gen/node_modules
```

Comment:
I think `promgen/static/primevue/` should also be added to `.gitignore`, since it will be generated by the build pipeline. It will be included in the released tarballs, but it should not be commit into the repo.

Reply:
🤔  I think we don't need to add that to `.gitignore`, since we only use this command to generate file only when we want to import new components.

Reply:
Generally I would agree that generated files should not be added, but should we consider the case of a python developer who wanted to work on the django bits, but not touch the javascript bits. Should we require them to also have node installed to get a working environment to develop in 🤔 
I think I'm slightly in favor of committing the generated files to `promgen/static/primeview` similar to how we have the binary files from vue and bootstrap and such downloaded from a release archive 🤔

Reply:
> should we consider the case of a python developer who wanted to work on the django bits, but not touch the javascript bits. Should we require them to also have node installed to get a working environment to develop in 

I don't think requiring to install node on a dev machine is something necessarily bad. It is a dev machine after all.

> I think I'm slightly in favor of committing the generated files to promgen/static/primeview similar to how we have the binary files from vue and bootstrap and such downloaded from a release archive 🤔

I get your point, but I'm more in favor of not committing the generated files for the following reasons:
- The Vue and Bootstrap libraries that we include come from official releases of those projects. They are public and well known, and they are not subject to change unless we (internally) decide to update to new releases. Nobody will ever modify those libraries that we include in our repo.
- The generated `main.css` and `main.js` files do not come from a trusty source. Instead, they are the result of a command line execution. And they are subject to change by anyone. For instance, I could imagine someone from anywhere in the world doing a PR for changes in the UI, which involve a new import for a Primevue component that we do not use yet. That would modify the generated `main.js`, which is around 7000 lines of JavaScript. Reviewing that would not hard. In fact, if we end up minifying that file, which I think we should do for performance reasons, it would be not just hard but impossible.
- The generated `main.css` and `main.js` file do not just bundle code from the Primevue library. Right now they do, yes. But in the future they will also bundle compiled single file component (`.vue`) files. Reviewing such compiled JavaScript code is hard or impossible, and the PRs could come from anywhere (untrusted sources), so they should be reviewed. Who knows what kind of bad stuff could be injected in an unreviewable soup of generated JavaScript code.

If you really want to go with committing those files, fine, I would not object more. But if I put those reasons on the left plate of a scale, and installing node on the right plate, I would choose the right plate.

Reply:
I agree that builded codes from others are untrusted.
If we have to be considerate of the developer who don't want to set up node environment, we have a mean that we build `main.css` and `main.js` by us in advance I think.

Reply:
The python developer in me doesn't like the idea of requiring node just to run Promgen for development, but to avoid issues with old binary versions and untrusted binaries in pull requests, it's likely best that we don't check in the build outputs and require node for development environments as well.

- We need to ensure that the Makefile properly checks requirements (`make run` should probably ensure the javascript files exist)
- We need to ensure that the docker build creates the required javascript files

Reply:
add this comments at [e699653](https://github.com/line/promgen/pull/456/commits/e699653ff33c1b70692fb25dc1a9a340aa13bc6f)